components/SCDDetect.coffee

```diff
@@ -8,25 +8,34 @@ exec = require('child_process').exec
 # @name SCDDetect
 
 compute = (canvas, cascade, callback) ->
```

Comment:
Does not need / should not be changed now, but how `compute()` relates to `onEnd` is pretty bad. Better if:

Almost all of of functionionality in `compute` is moved to a function `writeCanvasTempfile = (canvas, callback) ->` which will `callback Error|null, tempFile` (never calls into onEnd directly). This then becomes generic, and can potentially be reused other places.
`onEnd` should be renamed to something like `runScdDetect`; functions should be named after what they do, not how they are used - because function should not know/care how it is used, the caller decides that.
If a `compute()` function is still desired, it now becomes something like:
```
compute = (canvas, cascade, callback) ->
  writeCanvasTempfile canvas, (err, tmpFile) ->
    return callback err if err
    runScdDetect tmpFile, cascade, callback
```
But this is now minimal enough that one potentially do it inside the component.